### PR TITLE
[Bugfix]Fix bug truncate partition bucketNum following table not previous partition

### LINKED ISSUE #10435: truncate partition bucketNum following table not previous partition
### Steps to reproduce the behavior (Required)
1.
CREATE TABLE `bucket1` (
  `day` date NOT NULL COMMENT "",
  `date_time` datetime NOT NULL COMMENT "",
  `product` varchar(64) NOT NULL COMMENT ""
) ENGINE=OLAP 
UNIQUE KEY(`day`, `date_time`)
COMMENT "OLAP"
PARTITION BY RANGE(`day`)
(
PARTITION p20220630 VALUES [('2022-06-30'), ('2022-07-01')),
PARTITION p20220701 VALUES [('2022-07-01'), ('2022-07-02')),
PARTITION p20220702 VALUES [('2022-07-02'), ('2022-07-03')),
PARTITION p20220703 VALUES [('2022-07-03'), ('2022-07-04')),
PARTITION p20220704 VALUES [('2022-07-04'), ('2022-07-05')),
PARTITION p20220705 VALUES [('2022-07-05'), ('2022-07-06')))
DISTRIBUTED BY HASH(`date_time`) BUCKETS 10 
PROPERTIES (
"replication_num" = "3",
"dynamic_partition.enable" = "true",
"dynamic_partition.time_unit" = "DAY",
"dynamic_partition.time_zone" = "Asia/Shanghai",
"dynamic_partition.start" = "-21",
"dynamic_partition.end" = "3",
"dynamic_partition.prefix" = "p",
"dynamic_partition.buckets" = "30",
"dynamic_partition.replication_num" = "3",
"in_memory" = "false",
"storage_format" = "DEFAULT"
);
2.ALTER TABLE bucket1
SET ("default_partition.buckets" = "30");
3.truncate table bucket1 PARTITION(p20220707)

### Expected behavior (Required)
p20220707 Buckets is 30 (following old partition)
### Real behavior (Required)
p20220707 Buckets is 10 (following old table)
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
2.1.13

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [x] bugfix
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #10435

## Problem Summary(Required) ：

The truncate partition is now the bucketNum on the table, and now it is changed to the distribution of the old partition before.
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 8 / 8 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|8|8|100.00%|[]|

**Review by gengjun-git — DISMISSED** [review]

**Review by meegoo — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by Astralidea** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 1]
```diff
@@ -4169,8 +4171,8 @@ public void truncateTable(TruncateTableStmt truncateTableStmt) throws DdlExcepti
             }
 
             // check partitions
-            for (Map.Entry<String, Long> entry : origPartitions.entrySet()) {
-                Partition partition = copiedTbl.getPartition(entry.getValue());
+            for (Map.Entry<String, Partition> entry : origPartitions.entrySet()) {
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:4174]: Add a unit test to test it.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 3 / 6 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|3|6|50.00%|[4082, 4174, 4175]|

**Review by gengjun-git — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Pull request merged**